src/main/kotlin/uk/gov/dluhc/printapi/database/repository/CertificateRepository.kt

```diff
@@ -59,10 +63,30 @@ object CertificateRepositoryExtensions {
         )
     }
 
-    fun CertificateRepository.findPendingRemovalOfFinalRetentionData(sourceType: SourceType): List<Certificate> {
+    fun CertificateRepository.findPendingRemovalOfFinalRetentionData(
+        sourceType: SourceType,
+        batchIndex: Int,
+        batchSize: Int = 10000
+    ): Page<CertificateRemovalSummary> {
+        val batchRequest = PageRequest.of(
+            batchIndex,
+            batchSize,
+            Sort.by(Sort.Direction.ASC, "issueDate")
```

Comment:
shouldn't the sort by field be `final_retention_removal_date` so that it can use the index.
if you added the new s3arn column that Vishal is adding to the index certificate_final_retention_removal_idx then the query should never even load the table's data making it pretty fast.

Reply:
The `final_retention_removal_date` will be the same for each entry, so using `issueDate` is the better option.

Good point about re index. @vishalgupta-valtech - would you mind modifying it in your liquibase script? 😄

Reply:
Index created to include `id` 
<img width="1581" alt="image" src="https://user-images.githubusercontent.com/73763010/224097900-4cf73d80-2429-4792-a884-f1ca5e811ac5.png">

Reply:
Thank you 🙌  I've changed the `Sort.by` column in the repository 👍